File: backend/schemas/ohlc.py
from datetime import date
from typing import List

from pydantic import BaseModel


class OHLCPoint(BaseModel):
    date: str           # ISO "YYYY-MM-DD" — lightweight-charts expects string dates
    open: float
    high: float
    low: float
    close: float
    volume: float
    turnover: float | None = None
    market_cap: float | None = None

    model_config = {"from_attributes": True}


class OHLCMeta(BaseModel):
    symbol: str
    company_name: str | None = None
    interval: str
    total_records: int
    start_date: str
    end_date: str
    latest_close: float | None = None
    change: float | None = None
    change_pct: float | None = None


class OHLCResponse(BaseModel):
    symbol: str
    company_name: str | None = None
    interval: str
    ohlc: List[OHLCPoint]

File: frontend/src/types/ohlc.ts
export interface OHLCPoint {
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  turnover?: number;
  market_cap?: number;
}

export interface OHLCResponse {
  symbol: string;
  company_name?: string;
  interval: string;
  ohlc: OHLCPoint[];
}

export interface StandardResponse<T> {
  success: boolean;
  data: T | null;
  meta?: {
    total: number;
    page: number;
    page_size: number;
    total_pages: number;
  } | null;
  errors?: string | null;
}

export type Interval = 'daily' | 'weekly' | 'monthly';
export type QuickRange = '1M' | '3M' | '6M' | 'YTD' | '1Y' | '2Y' | '5Y' | 'MAX';
export type ChartType = 'candlestick' | 'line' | 'area';
